### PR TITLE
Lint against lowercase static mut

### LINKED ISSUE #37145: Naming convention for static mut is not enforced
The following program does not report any naming convention warnings:

```rust
#![allow(dead_code)]

static mut snake_case: u32 = 0;
static mut camelCase: u32 = 0;
static mut SCREAMING_SNAKE_CASE: u32 = 0;
static mut CapitalizedCamelCase: u32 = 0;

fn main() {}
```

Looks like only `SCREAMING_SNAKE_CASE` should be allowed? If I remove `mut`s, I get the warnings. Or have we already deprecated static muts (pre rfc: https://internals.rust-lang.org/t/pre-rfc-remove-static-mut/1437/20) ?

### PR DESCRIPTION
**Pull request by matklad**
Closes #37145.

Lint for non mut statics was added in https://github.com/rust-lang/rust/pull/7523, and it explicitly did not cover mut statics. I am not sure why.

### CODE REVIEW

File: src/librustc_lint/bad_style.rs [hunk 1]
```diff
@@ -355,10 +355,12 @@ impl LintPass for NonUpperCaseGlobals {
 impl LateLintPass for NonUpperCaseGlobals {
     fn check_item(&mut self, cx: &LateContext, it: &hir::Item) {
         match it.node {
-            // only check static constants
             hir::ItemStatic(_, hir::MutImmutable, _) => {
                 NonUpperCaseGlobals::check_upper_case(cx, "static constant", it.name, it.span);
             }
+            hir::ItemStatic(_, hir::MutMutable, _) => {
+                NonUpperCaseGlobals::check_upper_case(cx, "static variable", it.name, it.span);
+            }
```
Comment (jseyfried) [inline src/librustc_lint/bad_style.rs]: I believe the above two arms can be merged into a single arm:
```rust
    hir::ItemStatic(..) => {
        NonUpperCaseGlobals::check_upper_case(cx, "static variable", it.name, it.span);
    }
```

**Review by jseyfried — COMMENTED** [review]
Thanks! r=me with the comment addressed.

**Comment by matklad** [comment]
This may produce new warnings. Should the relnotes label be applied?

**Comment by alexcrichton** [comment]
Yeah, let's tag

**Pull request merged**